In [1]:
import pandas as pd
import numpy as np

oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(oceny)

0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


Zadanie 1a:

In [2]:
pensje = pd.Series({
    'Anna' : 5500,
    'Jan' : 7200,
    'Ewa' : 4800,
    'Marek' : 9100,
    'Kasia' : 6300
})
print(pensje)

Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64


Zadanie 1b

In [4]:
print(f"Najwięcej zarabia: {pensje.idxmax()}")
print(f"Średnia pensja: {pensje.mean()}")
print(f"Powyżej średniej:\n{pensje[pensje > pensje.mean()]}")
print(f"Poniżej 6000: {(pensje < 6000).sum()} osoby")


Najwięcej zarabia: Marek
Średnia pensja: 6580.0
Powyżej średniej:
Jan      7200
Marek    9100
dtype: int64
Poniżej 6000: 2 osoby


Zadanie 1c


In [5]:
po_podwyzce = pensje * 1.1
po_premii = pensje + 500
nowa_pensja = po_podwyzce + 500
print(f"Ranking nowych pensji:")
print(nowa_pensja.sort_values(ascending=False))


Ranking nowych pensji:
Marek    10510.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0
dtype: float64


Ćwiczenie 2 -DataFrame

In [6]:
# Dane o produktach sklepu internetowego
sklep = pd.DataFrame({
'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
'cena': [3500, 1800, 2500, 350, 2200],
'stan_magazynowy': [45, 120, 200, 500, 75],
'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})

Zadanie 2a


In [7]:
print(f"Kształt: {sklep.shape}")
print(f"Kolumny: {sklep.columns.tolist()}")
print(f"\nTypy danych:\n{sklep.dtypes}")
print(f"\nPierwsze 3 wiersze:\n{sklep.head(3)}")

Kształt: (5, 4)
Kolumny: ['produkt', 'cena', 'stan_magazynowy', 'kategoria']

Typy danych:
produkt              str
cena               int64
stan_magazynowy    int64
kategoria            str
dtype: object

Pierwsze 3 wiersze:
    produkt  cena  stan_magazynowy  kategoria
0    Laptop  3500               45  Komputery
1    Tablet  1800              120    Mobilne
2  Smartfon  2500              200    Mobilne


Zadanie 2b

In [8]:
sklep['wartosc_magazynu'] = sklep['cena'] * sklep['stan_magazynowy']
sklep['cena_z_vat'] = sklep['cena'] * 1.23
print(sklep)

     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0


Zadanie 2c

In [9]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)

print(f"Shape: {tips.shape}")
print(f"\nOstatnie 5 wierszy:\n{tips.tail()}")
print(f"\n3 losowe wiersze:\n{tips.sample(3)}")
print(f"\nTypy danych:\n{tips.dtypes}")

Shape: (244, 7)

Ostatnie 5 wierszy:
     total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

3 losowe wiersze:
     total_bill   tip     sex smoker   day    time  size
142       41.19  5.00    Male     No  Thur   Lunch     5
111        7.25  1.00  Female     No   Sat  Dinner     1
7         26.88  3.12    Male     No   Sun  Dinner     4

Typy danych:
total_bill    float64
tip           float64
sex               str
smoker            str
day               str
time              str
size            int64
dtype: object


Ćwiczenie 3

In [10]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)

Zadanie 3a

In [11]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)

print(penguins.head(10))
print(f"\nShape: {penguins.shape}")
penguins.info()

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       36

Zadanie 3b

In [12]:
print(penguins.describe())
print("\n--- include all ---")
print(penguins.describe(include='all'))
print(f"\nMediana masy ciała: {penguins['body_mass_g'].median()} g")

       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

--- include all ---
       species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      342.000000     342.000000         342.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       152     168             NaN            NaN 

Zadanie 3c

In [13]:
print(f"Gatunki:\n{penguins['species'].value_counts()}")
print(f"\nWyspy:\n{penguins['island'].value_counts()}")
print(f"\nPłeć:\n{penguins['sex'].value_counts()}")
print(f"\nProcent gatunków:\n{penguins['species'].value_counts(normalize=True).round(3) * 100}")

Gatunki:
species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

Wyspy:
island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

Płeć:
sex
MALE      168
FEMALE    165
Name: count, dtype: int64

Procent gatunków:
species
Adelie       44.2
Gentoo       36.0
Chinstrap    19.8
Name: proportion, dtype: float64


Zadanie 3d

In [14]:
braki = penguins.isna().sum()
print(f"Brakujące wartości:\n{braki}")

braki_sex = penguins['sex'].isna().sum()
procent = braki_sex / len(penguins) * 100
print(f"\nBraki w 'sex': {braki_sex} ({procent:.1f}%)")

Brakujące wartości:
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

Braki w 'sex': 11 (3.2%)


## Wnioski z analizy datasetu Penguins

1. Dataset zawiera 344 pingwiny z 3 gatunków zamieszkujących 3 wyspy Antarktyki.
2. Najczęstszy gatunek to Adelie — 152 osobniki, co stanowi 44.2% wszystkich obserwacji.
3. Brakujące dane występują w 4 kolumnach, najwięcej w kolumnie 'sex' — 11 braków (3.2%).
4. Średnia masa ciała pingwinów wynosi ok. 4202 g, a mediana to 4050 g.
5. Pingwiny zamieszkują 3 wyspy, najliczniej wyspę Biscoe — mieszka tam gatunek Gentoo.

Ćwiczenie 4


Zadanie 4a

In [17]:
tips = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv')
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

print(f"Średni rachunek: {tips['total_bill'].mean():.2f} zł")
print(f"Średni napiwek %: {tips['tip_pct'].mean():.1f}%")
print(f"\nNajwięcej rachunków wg dnia:\n{tips['day'].value_counts()}")
print(f"\nPalacze:\n{tips['smoker'].value_counts()}")
print(f"\nStatystyki napiwku:\n{tips['tip'].describe()}")
print(f"\nRozkład posiłków:\n{tips['time'].value_counts(normalize=True).round(3) * 100}")
print(f"\nUnikalne rozmiary grup: {tips['size'].nunique()}")

Średni rachunek: 19.79 zł
Średni napiwek %: 16.1%

Najwięcej rachunków wg dnia:
day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64

Palacze:
smoker
No     151
Yes     93
Name: count, dtype: int64

Statystyki napiwku:
count    244.000000
mean       2.998279
std        1.383638
min        1.000000
25%        2.000000
50%        2.900000
75%        3.562500
max       10.000000
Name: tip, dtype: float64

Rozkład posiłków:
time
Dinner    72.1
Lunch     27.9
Name: proportion, dtype: float64

Unikalne rozmiary grup: 6
